src/main/scala/ClockUtil.scala

```diff
@@ -164,6 +164,26 @@ class ClockDivider(width: Int) extends Module {
 
 }
 
+class ClockDivideAndMux(width: Int, depth: Int = 3, genClockGate: () => ClockGate) extends Module {
```

Comment:
Do you care about stopping the clock? Why not just make this be the divider on `divisor = [1, N)` and the passthrough on `divisor = 0`? No subtract operation needed?

Reply:
Good point. I actually prefer this way. If we want to stop the clock, we can use the existing clock gating stuff.